File: websocket-server/src/main/java/com/jonathan/model/Message.java
package com.jonathan.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Message {

    private String username;
    private String content;

//    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
    private LocalDateTime date;
    private String type;
    private List<String> userList;

    public static class Type{
        public static final String USER_IN = "USER_IN";
        public static final String MESSAGE = "MESSAGE";
        public static final String USER_OUT = "USER_OUT";
        public static final String IN_OK = "IN_OK";
        public static final String IN_REQ = "IN_REQ";
    }
}

File: websocket-client/src/components/chat/Message.js
export const USER_IN = "USER_IN"
export const USER_OUT = "USER_OUT"
export const MESSAGE = "MESSAGE"
export const USER_LIST = "USER_LIST"
export const IN_OK = "IN_OK"
export const IN_REQ = "IN_REQ"

const Message = (props) => {

    // const formatDate = () => {
    //     let date = props.message.date
    //     return date.slice(0, 10)+", "+date.slice(11, -5)
    // }

    switch(props.message.type){
        case USER_IN:
            return (
                <div className="userIn">
                    <p>{props.message.username} joined.</p>
                </div>
            )
        case MESSAGE:
            return (
                props.message.username !== props.username ?
                    <div className="messageOthers">
                        <p className="username">{props.message.username}</p>
                        <p className="content">{props.message.content}</p>
                        <p className="date">{props.message.date}</p>
                    </div> :
                    <div className="messageMe">
                        <p className="content">{props.message.content}</p>
                        <p className="date">{props.message.date}</p>
                    </div>
            )
        case USER_OUT:
            return (
                <div className="userOut">
                    <p>{props.message.username} left chat.</p>
                </div>
            )
        default:
            return(
                <div>default case. you shouldn't see this!</div>
            )
    }
}

export default Message